### ISSUE TITLE
Icons and text are messed up in categories

### ISSUE THREAD

**Issue by pvalsecc** [keepassxreboot/keepassxc#5015]
## Overview
When editing an entry or in the application settings the text below the icons on the left is impossible to read

## Steps to Reproduce
1. Linux with a HD screen, I guess

## Expected Behavior
I should see the full text height

## Actual Behavior
See screenshot:
![image](https://user-images.githubusercontent.com/1180608/87171843-8d232480-c2d3-11ea-8410-f2a75c6bccab.png)

## Context
Linux Ubuntu 20.04 with i3wm as window manager and a screen like that:
```
screen #0:
  dimensions:    3840x2160 pixels (712x400 millimeters)
  resolution:    137x137 dots per inch
```

KeePassXC - Version 2.6.0
Revision: 0765954

Qt 5.12.8
Debugging mode is disabled.

Operating system: Ubuntu 20.04 LTS
CPU architecture: x86_64
Kernel: linux 5.4.0-40-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.5

Desktop Env: i3
Windowing System: X11

**Comment by okalm** [comment]
I can confirm, in my case the first dialogue window is mostly unreadable, or some part don't show up properly  as you can see in pictures.
Monitor full Hd 1920*1080, screen dpi 92.

![bug_2_keepassxc](https://user-images.githubusercontent.com/40634668/87223579-7685d780-c37e-11ea-83e4-3f727e64be0d.png)
![bug_1_keepassxc](https://user-images.githubusercontent.com/40634668/87223656-0b88d080-c37f-11ea-8877-b664ece1d245.png)


Desktop Environment: i3wm with X11
GTK theme: Adwaita
Icon theme: Adwaita

KeePassXC - Version 2.6.0
Revision : 0765954

Qt 5.15.0
Debugging mode is disabled.

Operating system : Arch Linux
CPU architecture :  x86_64
Kernel : linux 5.7.7-arch1-1

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.6

**Comment by droidmonkey** [comment]
You can solve your problems with these env vars: https://doc.qt.io/qt-5/highdpi.html#high-dpi-support-in-qt

**Comment by okalm** [comment]
Thank you @droidmonkey , it worked !

I put the environment variable `export QT_SCALE_FACTOR=1.25` (I prefer when it's a little bigger but for most people `=1` should be enough)  in a the file `~/.xprofile` (not in `~/.bash*` files otherwise it works only when keepassxc is launched from the terminal) and that did the trick ;)

**Comment by pvalsecc** [comment]
If I change those variables, it messes up the layout even more and doesn't fix the icons' text being cut.

**Comment by droidmonkey** [comment]
Did you use KeePassXC before 2.6.0?

**Comment by droidmonkey** [comment]
@pvalsecc your problem is incredibly unique and not something seen before. Please try running version 2.5.4 and tell me if the categories are still messed up for you.

**Comment by orbrs** [comment]
I'm having the same issue as @okalm. It works when `QT_SCALE_FACTOR=1.05` or higher, but `1.0` and below makes the text all but disappear. I'm not using a QPlatformTheme if that's relevant.

![Screenshot-2020 07 14 1](https://user-images.githubusercontent.com/49759478/87419985-fd77c180-c5c3-11ea-902b-36f7231a1d30.png)
![Screenshot-2020 07 14 2](https://user-images.githubusercontent.com/49759478/87419995-01a3df00-c5c4-11ea-88e9-d1d60f4b49f4.png)

**Comment by droidmonkey** [comment]
Try setting `QT_SCALE_FACTOR_ROUNDING_POLICY=Round` with no other env var changes.

**Comment by orbrs** [comment]
> Try setting `QT_SCALE_FACTOR_ROUNDING_POLICY=Round` with no other env var changes.

Yes perfect! That's fixed it. Thanks @droidmonkey.

I'm also on Arch Linux and the previous version, 2.5.4, didn't have this problem - if that helps narrow down the root cause.

**Comment by droidmonkey** [comment]
That's the cause, and it only affects HD resolution displays, perhaps ones with non standard DPI. There are open bug reports in Qt bug tracker regarding this.

**Comment by droidmonkey** [comment]
Going to be fixed in the same way as #5029 will be fixed.
